### ISSUE TITLE
Search history for Chinese

### ISSUE THREAD

**Issue by FranklinYu** [overtake/TelegramSwift#451]
Current search logic seems English-centric. For example, words are separated by spaces, and the search term needs to match an entire word. This doesn’t work for Chinese. (Actually I guess it doesn’t work for Japanese or Korean, either.)

I think it would be helpful to have different word-splitting logic for Chinese, Japanese, and Korean.

**Comment by JeremyLIU1997** [comment]
Absolutely. Just to elaborate a little more in case the situation is not clear though:
e.g. given a message: 你好，世界！
Searching for 你好 or 世界 will work, but searching for 你 or 好 or 世 or 界 wouldn't, because the search string has to matching exactly what's between spaces/punctuation.

Hope this can be fixed. Thank you!
